### PR TITLE
add modal react previews in patterns section

### PR DESCRIPTION
**Pull request by hwigmore**
[Clubhouse](https://app.clubhouse.io/floored/story/10823/add-react-modal-previews-to-documentation)

@emma-floored i know these haven't been officially spec'd out yet, but since they exist and since we use them, it felt like we needed a react preview as a start. 

liz thought Patterns was the right place for them

### CODE REVIEW

File: react/modals/ActionModal.jsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ const ButtonSecondary = require('../buttons/ButtonSecondary.jsx');
 
 
 const ActionModal = (props) => {
+  if (props.isHidden) return null;
```
Comment (hwigmore) [inline react/modals/ActionModal.jsx]: `BaseModal` has `props.children` as required, so we were getting an error how this was (making `modalContent = null`). i figured this is clearer anyway and saves a few steps below!

File: docs/patterns/modals.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+layout: componentSection
+section: Components
```
Comment (emma-floored) [inline docs/patterns/modals.html]: change to "Patterns"?

Comment (emma-floored) [reply docs/patterns/modals.html]: oh actually no, the Tabs page also still says "Components"... basically when I run this, the modals link doesn't show up under "Patterns", and I'm trying to figure out why

Comment (emma-floored) [reply docs/patterns/modals.html]: But it does seem to be fixed if you change this to "Patterns". But if you update it, could you also update it on tabs.html ?? 💐 💐 💐

**Comment by emma-floored** [comment]
other than my one series of comments, this looks good! 👍

File: docs/patterns/modals.html (hunk 2 above)
Comment (hwigmore) [reply docs/patterns/modals.html]: oh i copied and pasted this file from tabs yeah! will update both!

**Pull request merged**